sale_automatic_workflow_with_delay/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    @api.model
+    def run_with_workflow_with_delay(self, sale_workflow):
+        workflow_domain = [("workflow_process_id", "=", sale_workflow.id)]
+        if sale_workflow.validate_order:
+            self.with_context(
+                send_order_confirmation_mail=sale_workflow.send_order_confirmation_mail,
+            )._validate_sale_orders(
+                safe_eval(sale_workflow.order_filter_id.domain) + workflow_domain
+            )
+
+    def run_with_workflow(self, sale_workflow):
+        super_self = self.with_context(
+            sale_workflow_validate_with_delay=sale_workflow.validate_order_with_delay
+        )
+        return super(AutomaticWorkflowJob, super_self).run_with_workflow(sale_workflow)
+
+    def _validate_sale_orders(self, order_filter):
+        if self.env.context.get("sale_workflow_validate_with_delay"):
+            if not self.env.context.get("sale_workflow_job_delayed"):
```

Comment:
this key is not set anywhere, am I wrong?

Reply:
indeed, you are right.